modules/client/examples/01-client/16-is-dao-update-proposal-valid.ts

```diff
@@ -0,0 +1,37 @@
+/* MARKDOWN
+---
+title: Check if a proposal valid for updating a dao
+---
+
+### Check if a proposal valid for updating a dao
+
+It will check that the proposal has the correct actions to update a dao.
+It will check that the eencoded previous version matches the dao version
+it will check that the implementation address matches the latest dao factory address
+It will check that the init data is empty becuase the dao factory does not have new params so init data is not necessary
+
+*/
+
+import { Client } from "@aragon/sdk-client";
+import { context } from "../index";
+
+// Instantiate the general purpose client from the Aragon OSx SDK context.
+const client: Client = new Client(context);
+
+const proposalId =
+  "0x1234567890123456789012345678901234567890123456789012345678901234";
+
+// Check proposal validity.
+const daoUpdateValidity = client.methods.isDaoUpdateProposalValid(proposalId);
+console.log(daoUpdateValidity);
+/* MARKDOWN
+    Returns:
+    ```tsx
+    {    
+        version: true,
+        implementation: true,
+        initData: true,
+        actions: true
+    }
+    ```
+    */
```

Comment:
This imply that this method can only be used if a proposal is already created, 
And can not be used inside the proposal creation flow, is that ok @jamesej @Michael-A-Heuer ?

Reply:
I saw the figma and I only saw this been used when the proposal is alredy done, I can implement it the other way around if you prefer it

Reply:
it is not my preference, when i looked at the code i thought of this, so i wanted to raise the point

Reply:
Yeah it's fine for this to be only for existing proposals. When we create a proposal for updates, all the calldata is automatically generated so we can be sure it's valid. This check is to warn the user if an update proposal might have been created outside the app and be malicious or incorrect.